### PR TITLE
SqlAGDatabase: Fix impersonate permission test

### LINKED ISSUE #1237: SqlAGDatabase: Impersonate permission incorrectly checked on SQL 2012
#### Details of the scenario you tried and the problem that is occurring
When running SqlAGDatabase with the MatchDatabaseOwner option Test-ImpersonatePermissions gets called. This dumps the fn_my_permissions for the user and checks for IMPERSONATE ANY LOGIN.

That permission does not exist on SQL 2012 and was only introduced in SQL 2014. This causes the test to fail even when equivalent permissions exist.

#### Verbose logs showing the problem
```
PowerShell DSC resource MSFT_SqlAGDatabase  failed to execute Set-TargetResource functionality with error message: The
login 'LAB\LocalAdministrator' is missing impersonate permissions in the instances 'DAC1N1, SEC1N2, DAC1N2, SEC1N3'.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : ProviderOperationExecutionFailure
    + PSComputerName        : SEC1N1
```

#### Suggested solution to the issue
- Check for CONTROL SERVER permission. This is given directly or through sysadmin, and grants an implicit IMPERSONATE ANY LOGIN permission.
- Check for IMPERSONATE LOGIN permission for each database owner.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
                        SqlAGDatabase "AddDatabaseTo$($node.Role.AvailabilityGroup.Name)"
                        {
                            AvailabilityGroupName   = $node.Role.AvailabilityGroup.Name
                            BackupPath              = '\\CHDC1\Temp' # TODO: Remove this
                            DatabaseName            = "Dummy$($node.Role.AvailabilityGroup.Name)"
                            ServerName              = $node.NodeName
                            InstanceName            = $node.Role.SQLServer.InstanceName
                            Ensure                  = 'Present'
                            PsDscRunAsCredential    = $localAdministrator
                            MatchDatabaseOwner = $true # EXECUTE AS
                            DependsOn = '[WaitForAll]WaitForAllAGReplicas'
                        }
```

#### SQL Server edition and version the target node is running
SQL 2012

#### SQL Server PowerShell modules present on the target node
SqlServer, SQLPS

#### The operating system the target node is running
Windows Server 2012

#### Version and build of PowerShell the target node is running
5.1

#### Version of the DSC module that was used ('dev' if using current dev branch)
dev

### PR DESCRIPTION
**Pull request by codykonior**
#### Pull Request (PR) description
When adding a database to an AG using SqlAGDatabase and the MatchDatabaseOwner switch, it checks for an IMPERSONATE ANY LOGIN permission.

That permission doesn't exist < SQL 2014. There are two alternatives:
- CONTROL SERVER (also granted as part of the sysadmin server role) permission
- IMPERSONATE <LoginName> permission

This change adds those two checks. It also fixes the help text that said MatchDatabaseOwner is true by default, when it was always false by default.

#### This Pull Request (PR) fixes the following issues
- Fixes #1237 

#### Task list
- [X] Added an entry under the Unreleased section of the change log in the CHANGELOG.md.
      Entry should say what was changed, and how that affects users (if applicable).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [X] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/sqlserverdsc/1239)

### CODE REVIEW

**Comment by codykonior** [comment]
The CI failure seem to be unrelated to my CR.

I can attempt to deal with the codecov/patch stuff though.

**Comment by johlju** [comment]
@codykonior Yes, it seems there is a intermittent error that occurs sometimes. Hopefully it works on the next push. It is also possible to kick off the tests again by closing and then reopening this PR. But let's wait for next push if you can look at getting more coverage of the changed code.

**Comment by johlju** [comment]
@codykonior I wait to review this until you resolved the missing code coverage. 🙂

**Comment by codykonior** [comment]
That second build failed for unrelated reasons again but this one passed. 

I haven’t been able to find the last few % of code coverage, the reporting doesn’t seem to indicate where it is. 

I put in some extra properties to the AGDatabase test file. These are needed if the tests are expanded to remove the Test-EffectivePermissions mock in favour of an Invoke-Query mock. I tried it out but didn’t modify the tests because it should already be covered, and doing so would create a dependency for that test on the SQLPS/SqlServer module that wasn’t there before (not sure why, sends it down more code paths I guess). 

So basically let me know if you know what’s not being tested and if you need more.

**Comment by codykonior** [comment]
Trying to rebase has broken this. I'll re-raise it.
